api/envoy/config/route/v3/route_components.proto

```diff
@@ -224,6 +224,13 @@ message FilterAction {
   google.protobuf.Any action = 1;
 }
 
+// This can be used in route matcher :ref:`VirtualHost.matcher<envoy_v3_api_field_config.route.v3.VirtualHost.matcher>`.
+// When the matcher matches, routes will be matched and run.
+message RouteMatchAction {
+  // The list of routes that will be matched and run, in order. The first route that matches will be used.
+  repeated Route routes = 1;
+}
```

Comment:
Please add a release note, also add a documentation example for this wherever we document how to use the matching system.